### ISSUE TITLE
In Windows ServerCore Container SSHD Works Interactively But Fails When Ran as a Service

### ISSUE THREAD

**Issue by aolszowka** [PowerShell/Win32-OpenSSH#1725]
**"OpenSSH for Windows" version**
8.1.0.0
 
**Server OperatingSystem**
Windows Server Datacenter - (Windows Server Core 20H2 via the Docker Windows Containers, I have included my dockerfile below).

**What is failing**
* Attempting to connect to the container via ssh when sshd is run via a service fails.
* Running sshd interatively (`C:\Program Files\OpenSSH\sshd.exe`) works with no issue

**Expected output**
I am expecting to be able to connect to ssh when this is run via the service.

**Actual output**
When attempting to ssh in when it is ran as a service:

```
ssh 172.29.11.62 -l ContainerAdministrator
ContainerAdministrator@172.29.11.62: Permission denied (publickey,keyboard-interactive).
```

Works without issue when sshd.exe is run interactively:

```
Microsoft Windows [Version 10.0.19042.746]
(c) 2020 Microsoft Corporation. All rights reserved.

user manager\containeradministrator@A644ECF04ADE C:\Users\ContainerAdministrator>
```

**dockerfile**
```
FROM mcr.microsoft.com/windows/servercore:20H2

ARG POWERSHELL_MSI=PowerShell-7.1.1-win-x64.msi # From https://github.com/PowerShell/PowerShell/releases
ARG OPENSSH_ZIP=OpenSSH-Win64-8.1.0.0p1-Beta.zip # From https://github.com/PowerShell/Win32-OpenSSH/releases

RUN MKDIR c:\\installers
COPY $POWERSHELL_MSI "C:\\installers\\PowerShell.msi"
RUN msiexec.exe /package C:\\installers\\PowerShell.msi /quiet ENABLE_PSREMOTING=1 REGISTER_MANIFEST=1

COPY $OPENSSH_ZIP "C:\\installers\\OpenSSH.zip"
RUN pwsh.exe -Command Expand-Archive -Path "C:\\installers\\OpenSSH.zip" -DestinationPath "C:\\"

# Install SSH as per https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH
RUN pwsh.exe -Command "& { Move-Item -Path 'C:\\OpenSSH-Win64' -Destination 'C:\\Program Files\\OpenSSH' }"
RUN pwsh.exe -ExecutionPolicy Bypass -File "C:\\Program Files\\OpenSSH\\install-sshd.ps1"

# Copy Our Configuration File
COPY "sshd_config" "C:\\ProgramData\\ssh\\sshd_config"

# Copy the authorized keys and set the permissions appropriately
COPY "administrators_authorized_keys" "C:\\ProgramData\\ssh\\administrators_authorized_keys"
RUN ICACLS C:\\ProgramData\\ssh /inheritancelevel:r
RUN ICACLS C:\\ProgramData\\ssh /remove *S-1-5-11
RUN ICACLS C:\\ProgramData\\ssh /grant SYSTEM:(F)
RUN ICACLS C:\\ProgramData\\ssh /grant BUILTIN\Administrators:(F)

RUN pwsh.exe -Command "& { Set-Service sshd -StartupType Automatic }"
RUN pwsh.exe -Command "& { Start-Service sshd }"

EXPOSE 22
```

**sshd_config**
```
# This configuration file is configured according to the following:
# https://docs.microsoft.com/en-us/powershell/scripting/learn/remoting/ssh-remoting-in-powershell-core?view=powershell-7.1#set-up-on-a-windows-computer

# This is the sshd server system-wide configuration file.  See
# sshd_config(5) for more information.

# The strategy used for options in the default sshd_config shipped with
# OpenSSH is to specify options with their default value where
# possible, but leave them commented.  Uncommented options override the
# default value.

#Port 22
#AddressFamily any
#ListenAddress 0.0.0.0
#ListenAddress ::

#HostKey __PROGRAMDATA__/ssh/ssh_host_rsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_dsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_ecdsa_key
#HostKey __PROGRAMDATA__/ssh/ssh_host_ed25519_key

# Ciphers and keying
#RekeyLimit default none

# Logging
SyslogFacility LOCAL0
LogLevel DEBUG3

# Authentication:

#LoginGraceTime 2m
PermitRootLogin prohibit-password
#StrictModes yes
#MaxAuthTries 6
#MaxSessions 10

PubkeyAuthentication yes

# The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
# but this is overridden so installations will only check .ssh/authorized_keys
AuthorizedKeysFile	.ssh/authorized_keys

#AuthorizedPrincipalsFile none

# For this to work you will also need host keys in %programData%/ssh/ssh_known_hosts
#HostbasedAuthentication no
# Change to yes if you don't trust ~/.ssh/known_hosts for
# HostbasedAuthentication
#IgnoreUserKnownHosts no
# Don't read the user's ~/.rhosts and ~/.shosts files
#IgnoreRhosts yes

# To disable tunneled clear text passwords, change to no here!
PasswordAuthentication no
#PermitEmptyPasswords no

# GSSAPI options
#GSSAPIAuthentication no

#AllowAgentForwarding yes
#AllowTcpForwarding yes
#GatewayPorts no
#PermitTTY yes
#PrintMotd yes
#PrintLastLog yes
#TCPKeepAlive yes
#UseLogin no
#PermitUserEnvironment no
#ClientAliveInterval 0
#ClientAliveCountMax 3
#UseDNS no
#PidFile /var/run/sshd.pid
#MaxStartups 10:30:100
#PermitTunnel no
#ChrootDirectory none
#VersionAddendum none

# no default banner path
#Banner none

# override default of no subsystems
Subsystem	sftp	sftp-server.exe
Subsystem powershell pwsh.exe -sshs -NoLogo

# Example of overriding settings on a per-user basis
#Match User anoncvs
#	AllowTcpForwarding no
#	PermitTTY no
#	ForceCommand cvs server

Match Group administrators
       AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
```

**DEBUG3 From the Service**
```
2368 2021-01-18 10:59:43.090 debug2: fd 3 setting O_NONBLOCK
2368 2021-01-18 10:59:43.090 debug3: sock_set_v6only: set socket 3 IPV6_V6ONLY
2368 2021-01-18 10:59:43.090 debug1: Bind to port 22 on ::.
2368 2021-01-18 10:59:43.090 Server listening on :: port 22.
2368 2021-01-18 10:59:43.090 debug2: fd 4 setting O_NONBLOCK
2368 2021-01-18 10:59:43.090 debug1: Bind to port 22 on 0.0.0.0.
2368 2021-01-18 10:59:43.090 Server listening on 0.0.0.0 port 22.
2368 2021-01-18 10:59:48.859 debug3: fd 5 is not O_NONBLOCK
2368 2021-01-18 10:59:48.859 debug3: spawning "C:\\Program Files\\OpenSSH\\sshd.exe" -R
2368 2021-01-18 10:59:48.859 debug3: send_rexec_state: entering fd = 8 config len 394
2368 2021-01-18 10:59:48.859 debug3: ssh_msg_send: type 0
2368 2021-01-18 10:59:48.859 debug3: send_rexec_state: done
1520 2021-01-18 10:59:48.874 debug1: inetd sockets after dupping: 4, 4
1520 2021-01-18 10:59:48.874 Connection from 172.29.0.1 port 56923 on 172.29.11.62 port 22
1520 2021-01-18 10:59:48.874 debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.1
1520 2021-01-18 10:59:48.874 debug1: Remote protocol version 2.0, remote software version OpenSSH_for_Windows_7.7
1520 2021-01-18 10:59:48.874 debug1: match: OpenSSH_for_Windows_7.7 pat OpenSSH* compat 0x04000000
1520 2021-01-18 10:59:48.874 debug2: fd 4 setting O_NONBLOCK
1520 2021-01-18 10:59:48.890 debug3: spawning "C:\\Program Files\\OpenSSH\\sshd.exe" -y
1520 2021-01-18 10:59:48.890 debug2: Network child is on pid 1964
1520 2021-01-18 10:59:48.890 debug3: send_rexec_state: entering fd = 6 config len 394
1520 2021-01-18 10:59:48.890 debug3: ssh_msg_send: type 0
1520 2021-01-18 10:59:48.890 debug3: send_rexec_state: done
1520 2021-01-18 10:59:48.890 debug3: ssh_msg_send: type 0
1520 2021-01-18 10:59:48.890 debug3: ssh_msg_send: type 0
1520 2021-01-18 10:59:48.890 debug3: preauth child monitor started
1520 2021-01-18 10:59:48.905 debug1: list_hostkey_types: rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
1520 2021-01-18 10:59:48.905 debug3: send packet: type 20 [preauth]
1520 2021-01-18 10:59:48.905 debug1: SSH2_MSG_KEXINIT sent [preauth]
1520 2021-01-18 10:59:48.905 debug3: receive packet: type 20 [preauth]
1520 2021-01-18 10:59:48.905 debug1: SSH2_MSG_KEXINIT received [preauth]
1520 2021-01-18 10:59:48.905 debug2: local server KEXINIT proposal [preauth]
1520 2021-01-18 10:59:48.905 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1 [preauth]
1520 2021-01-18 10:59:48.905 debug2: host key algorithms: rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
1520 2021-01-18 10:59:48.905 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
1520 2021-01-18 10:59:48.905 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
1520 2021-01-18 10:59:48.905 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
1520 2021-01-18 10:59:48.905 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
1520 2021-01-18 10:59:48.905 debug2: compression ctos: none,zlib@openssh.com [preauth]
1520 2021-01-18 10:59:48.905 debug2: compression stoc: none,zlib@openssh.com [preauth]
1520 2021-01-18 10:59:48.905 debug2: languages ctos:  [preauth]
1520 2021-01-18 10:59:48.905 debug2: languages stoc:  [preauth]
1520 2021-01-18 10:59:48.905 debug2: first_kex_follows 0  [preauth]
1520 2021-01-18 10:59:48.905 debug2: reserved 0  [preauth]
1520 2021-01-18 10:59:48.921 debug2: peer client KEXINIT proposal [preauth]
1520 2021-01-18 10:59:48.921 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c [preauth]
1520 2021-01-18 10:59:48.921 debug2: host key algorithms: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa [preauth]
1520 2021-01-18 10:59:48.921 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
1520 2021-01-18 10:59:48.921 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
1520 2021-01-18 10:59:48.921 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
1520 2021-01-18 10:59:48.921 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
1520 2021-01-18 10:59:48.921 debug2: compression ctos: none [preauth]
1520 2021-01-18 10:59:48.921 debug2: compression stoc: none [preauth]
1520 2021-01-18 10:59:48.921 debug2: languages ctos:  [preauth]
1520 2021-01-18 10:59:48.921 debug2: languages stoc:  [preauth]
1520 2021-01-18 10:59:48.921 debug2: first_kex_follows 0  [preauth]
1520 2021-01-18 10:59:48.921 debug2: reserved 0  [preauth]
1520 2021-01-18 10:59:48.921 debug1: kex: algorithm: curve25519-sha256 [preauth]
1520 2021-01-18 10:59:48.921 debug1: kex: host key algorithm: ecdsa-sha2-nistp256 [preauth]
1520 2021-01-18 10:59:48.921 debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
1520 2021-01-18 10:59:48.921 debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
1520 2021-01-18 10:59:48.921 debug1: expecting SSH2_MSG_KEX_ECDH_INIT [preauth]
1520 2021-01-18 10:59:48.921 debug3: receive packet: type 30 [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_sshkey_sign entering [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_request_send entering: type 6 [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_sshkey_sign: waiting for MONITOR_ANS_SIGN [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_request_receive_expect entering: type 7 [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_request_receive entering [preauth]
1520 2021-01-18 10:59:48.921 debug3: mm_request_receive entering
1520 2021-01-18 10:59:48.921 debug3: monitor_read: checking request 6
1520 2021-01-18 10:59:48.921 debug3: mm_answer_sign
1520 2021-01-18 10:59:48.921 debug3: mm_answer_sign: KEX signature 00000151F65183A0(100)
1520 2021-01-18 10:59:48.921 debug3: mm_request_send entering: type 7
1520 2021-01-18 10:59:48.921 debug2: monitor_read: 6 used once, disabling now
1520 2021-01-18 10:59:48.921 debug3: send packet: type 31 [preauth]
1520 2021-01-18 10:59:48.921 debug3: send packet: type 21 [preauth]
1520 2021-01-18 10:59:48.921 debug2: set_newkeys: mode 1 [preauth]
1520 2021-01-18 10:59:48.921 debug1: rekey out after 134217728 blocks [preauth]
1520 2021-01-18 10:59:48.921 debug1: SSH2_MSG_NEWKEYS sent [preauth]
1520 2021-01-18 10:59:48.921 debug1: Sending SSH2_MSG_EXT_INFO [preauth]
1520 2021-01-18 10:59:48.921 debug3: send packet: type 7 [preauth]
1520 2021-01-18 10:59:48.921 debug1: expecting SSH2_MSG_NEWKEYS [preauth]
1520 2021-01-18 10:59:48.921 debug3: receive packet: type 21 [preauth]
1520 2021-01-18 10:59:48.921 debug1: SSH2_MSG_NEWKEYS received [preauth]
1520 2021-01-18 10:59:48.921 debug2: set_newkeys: mode 0 [preauth]
1520 2021-01-18 10:59:48.921 debug1: rekey in after 134217728 blocks [preauth]
1520 2021-01-18 10:59:48.921 debug1: KEX done [preauth]
1520 2021-01-18 10:59:48.984 debug3: receive packet: type 5 [preauth]
1520 2021-01-18 10:59:48.984 debug3: send packet: type 6 [preauth]
1520 2021-01-18 10:59:48.984 debug3: receive packet: type 50 [preauth]
1520 2021-01-18 10:59:48.984 debug1: userauth-request for user ContainerAdministrator service ssh-connection method none [preauth]
1520 2021-01-18 10:59:48.984 debug1: attempt 0 failures 0 [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_getpwnamallow entering [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_send entering: type 8 [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_getpwnamallow: waiting for MONITOR_ANS_PWNAM [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_receive_expect entering: type 9 [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_receive entering [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_receive entering
1520 2021-01-18 10:59:48.984 debug3: monitor_read: checking request 8
1520 2021-01-18 10:59:48.984 debug3: mm_answer_pwnamallow
1520 2021-01-18 10:59:48.984 debug3: get_passwd: Invalid account type: 5.
1520 2021-01-18 10:59:48.984 debug2: parse_server_config: config reprocess config len 394
1520 2021-01-18 10:59:48.984 debug3: checking match for 'Group administrators' user ContainerAdministrator host 172.29.0.1 addr 172.29.0.1 laddr 172.29.11.62 lport 22
1520 2021-01-18 10:59:48.984 debug3: get_passwd: Invalid account type: 5.
1520 2021-01-18 10:59:48.984 debug1: Can't match group at line 91 because user ContainerAdministrator does not exist
1520 2021-01-18 10:59:48.984 debug3: match not found
1520 2021-01-18 10:59:48.984 Invalid user ContainerAdministrator from 172.29.0.1 port 56923
1520 2021-01-18 10:59:48.984 debug3: mm_answer_pwnamallow: sending MONITOR_ANS_PWNAM: 0
1520 2021-01-18 10:59:48.984 debug3: mm_request_send entering: type 9
1520 2021-01-18 10:59:48.984 debug2: monitor_read: 8 used once, disabling now
1520 2021-01-18 10:59:48.984 debug3: mm_inform_authserv entering [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_send entering: type 4 [preauth]
1520 2021-01-18 10:59:48.984 debug2: input_userauth_request: try method none [preauth]
1520 2021-01-18 10:59:48.984 debug3: mm_request_receive entering
1520 2021-01-18 10:59:48.984 debug3: monitor_read: checking request 4
1520 2021-01-18 10:59:48.984 debug3: mm_answer_authserv: service=ssh-connection, style=
1520 2021-01-18 10:59:48.984 debug2: monitor_read: 4 used once, disabling now
1520 2021-01-18 10:59:48.984 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
1520 2021-01-18 10:59:48.984 debug3: ensure_minimum_time_since: elapsed 0.000ms, delaying 6.343ms (requested 6.343ms) [preauth]
1520 2021-01-18 10:59:48.999 debug3: userauth_finish: failure partial=0 next methods="publickey,keyboard-interactive" [preauth]
1520 2021-01-18 10:59:48.999 debug3: send packet: type 51 [preauth]
1520 2021-01-18 10:59:48.999 debug3: receive packet: type 50 [preauth]
1520 2021-01-18 10:59:48.999 debug1: userauth-request for user ContainerAdministrator service ssh-connection method publickey [preauth]
1520 2021-01-18 10:59:48.999 debug1: attempt 1 failures 0 [preauth]
1520 2021-01-18 10:59:48.999 debug2: input_userauth_request: try method publickey [preauth]
1520 2021-01-18 10:59:48.999 debug2: userauth_pubkey: invalid user ContainerAdministrator querying public key ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAp/c1GFp4KqRo8P1wMMU+KRy7N+zXv2lrIAemAsZfUl [preauth]
1520 2021-01-18 10:59:48.999 debug1: userauth_pubkey: test pkalg ssh-ed25519 pkblob ED25519 SHA256:FLYfTBWUtmgmvjthkIwBSRv3Fp0bV4xpWddg4ShZBBc [preauth]
1520 2021-01-18 10:59:48.999 debug2: userauth_pubkey: disabled because of invalid user [preauth]
1520 2021-01-18 10:59:48.999 debug2: userauth_pubkey: authenticated 0 pkalg ssh-ed25519 [preauth]
1520 2021-01-18 10:59:48.999 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
1520 2021-01-18 10:59:48.999 debug3: ensure_minimum_time_since: elapsed 0.000ms, delaying 6.343ms (requested 6.343ms) [preauth]
1520 2021-01-18 10:59:49.015 debug3: userauth_finish: failure partial=0 next methods="publickey,keyboard-interactive" [preauth]
1520 2021-01-18 10:59:49.015 debug3: send packet: type 51 [preauth]
1520 2021-01-18 10:59:49.015 debug3: receive packet: type 50 [preauth]
1520 2021-01-18 10:59:49.015 debug1: userauth-request for user ContainerAdministrator service ssh-connection method keyboard-interactive [preauth]
1520 2021-01-18 10:59:49.015 debug1: attempt 2 failures 1 [preauth]
1520 2021-01-18 10:59:49.015 debug2: input_userauth_request: try method keyboard-interactive [preauth]
1520 2021-01-18 10:59:49.015 debug1: keyboard-interactive devs  [preauth]
1520 2021-01-18 10:59:49.015 debug1: auth2_challenge: user=ContainerAdministrator devs= [preauth]
1520 2021-01-18 10:59:49.015 debug1: kbdint_alloc: devices '' [preauth]
1520 2021-01-18 10:59:49.015 debug2: auth2_challenge_start: devices  [preauth]
1520 2021-01-18 10:59:49.015 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
1520 2021-01-18 10:59:49.015 debug3: ensure_minimum_time_since: elapsed 0.000ms, delaying 6.343ms (requested 6.343ms) [preauth]
1520 2021-01-18 10:59:49.030 debug3: userauth_finish: failure partial=0 next methods="publickey,keyboard-interactive" [preauth]
1520 2021-01-18 10:59:49.030 debug3: send packet: type 51 [preauth]
1520 2021-01-18 10:59:49.030 debug3: recv - from CB ERROR:10054, io:0000013BD36CDEF0 [preauth]
1520 2021-01-18 10:59:49.030 Connection reset by invalid user ContainerAdministrator 172.29.0.1 port 56923 [preauth]
1520 2021-01-18 10:59:49.030 debug1: do_cleanup [preauth]
1520 2021-01-18 10:59:49.030 debug1: monitor_read_log: child log fd closed
1520 2021-01-18 10:59:49.030 debug3: mm_request_receive entering
1520 2021-01-18 10:59:49.030 debug1: do_cleanup
1520 2021-01-18 10:59:49.030 debug1: Killing privsep child 1964
```

We can see from the log file that its failing to find the user for some reason:

```
1520 2021-01-18 10:59:48.984 debug3: mm_answer_pwnamallow
1520 2021-01-18 10:59:48.984 debug3: get_passwd: Invalid account type: 5.
1520 2021-01-18 10:59:48.984 debug2: parse_server_config: config reprocess config len 394
1520 2021-01-18 10:59:48.984 debug3: checking match for 'Group administrators' user ContainerAdministrator host 172.29.0.1 addr 172.29.0.1 laddr 172.29.11.62 lport 22
1520 2021-01-18 10:59:48.984 debug3: get_passwd: Invalid account type: 5.
1520 2021-01-18 10:59:48.984 debug1: Can't match group at line 91 because user ContainerAdministrator does not exist
```

I just cannot figure out why this is the case, unfortunately because this is in a container I am unable to use the suggested `psexec -s sshd.exe -d` to run sshd as SYSTEM to see if there is any better logging when ran that way.

**Comment by aolszowka** [comment]
So I jumped on discord with @jborean93.

TL;DR; - I ended up creating a throw away account with a random password and added it to the administrators group which fixed the issue.

**The Long**
`ContainerAdministrator` is the default account that comes out of the box with the Windows Containers produced by Microsoft. Apparently this account is a `SidTypeWellKnownGroup` which for some reason causes this to fail. Here's the raw data from the chat:

jborean93 2021/01/18 1:40 PM
https://github.com/PowerShell/openssh-portable/blob/8ab565c53f3619d6a1f5ac229e212cad8a52852c/contrib/win32/win32compat/pwd.c#L226-L230
refers to https://docs.microsoft.com/en-us/windows/win32/api/winnt/ne-winnt-sid_name_use
5 == SidTypeWellKnownGroup if I'm remembering that C enums start at 1

jborean93 2021/01/18 1:42 PM
yep the log entry is from https://github.com/PowerShell/openssh-portable/blob/latestw_all/contrib/win32/win32compat/pwd.c#L228
and account_type is defined as the large argument to LogonAccountSidW https://github.com/PowerShell/openssh-portable/blob/latestw_all/contrib/win32/win32compat/pwd.c#L218-L219
and that argument is peUse being SID_NAME_USE https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-lookupaccountsidw
5 is SidTypeWellKnownGroup https://github.com/tpn/winsdk-10/blob/master/Include/10.0.16299.0/um/winnt.h#L9026-L9038
so for whatever reason the account you are trying to connect with is seen as a well known group

Ace Olszowka 2021/01/18 1:48 PM
That would make sense, ContainerAdministrator is the default out of the box account for Windows Containers

jborean93 2021/01/18 1:49 PM
does it have a known password or are you trying to connect with pub key auth because it doesn't?

Ace Olszowka 2021/01/18 1:50 PM
The latter (trying to connect with pub key auth because we do not know the password)

Ace Olszowka 2021/01/18 1:50 PM
if we need to create a dummy account out of the box that is fine and set it to a random password
Yup that worked (adding a random account)
